test/test_sync.cpp

```diff
@@ -314,7 +314,8 @@ TEST(SyncEvent, WaitFor)
         // - SyncEvent::wait_for( 50us) took 6us
         // - SyncEvent::wait_for(100us) took 4us
         if (on_timeout) {
-            EXPECT_GE(waittime_us, timeout_us);
+            int tolerance = timeout_us/1000;
+            EXPECT_GE(waittime_us, timeout_us - tolerance);
```

Comment:
If `timeout_us` is 1 ms (1000 us), the condition would be 
```c++
EXPECT_GE(waittime_us, 0);
```
which does not check much.

Minor note: `const int tolerance`, even though the scope is within two lines of code.

Reply:
I don't understand. `timeout_us - tolerance = timeout_us - (timeout_us/1000)`.

So for 1ms `timeout_us == 1000` it's `1000 - (1000/1000) = 1000 - 1 = 999`.

And you're right about this const, of course, I'll add it.

Reply:
> So for 1ms timeout_us == 1000 it's 1000 - (1000/1000) = 1000 - 1 = 999.

Right, I lost units :)